### PR TITLE
Editorial review

### PR DESCRIPTION
**Pull request by gorryfair**
Please do not merge yet, I will tray to update to include a few sentences to address the issue related to in-network Circuit breakers.

### CODE REVIEW

File: draft-ietf-ccwg-rfc5033bis.md [hunk 1]
```diff
@@ -162,19 +162,19 @@ in an Informational Internet Draft in 2015, published as an
 Informational RFC in 2017 {{?RFC8312}} and then as a Proposed
 Standard in 2023 {{?RFC9438}}.
 
-BBR is developed as an internal research project by Google,
+BBR was developed as an internal research project by Google,
```
Comment (huitema) [inline draft-ietf-ccwg-rfc5033bis.md:165]: I think "is developed" is actually correct. There is ongoing active development, entirely at Google, and periodic reports to the IRTF/ICCRG or IETF/CCWG. Change control is firmly by Google, and there have been no attempt to make this an IETF project.

File: draft-ietf-ccwg-rfc5033bis.md [hunk 2]
```diff
@@ -346,6 +347,19 @@ full backoff mechanism must be identical to that of TCP
...
+that is used by some equipment in the network to continuously monitor a
+flow or aggregate set of flows.  The mechanism seeks to detect when the
+flow(s) experience persistent excessive congestion, and when detected,
+to terminate (or significantly reduces the rate of) the flow(s).
+
+A well-designed congestion control algorithm ought to react before
+a flow uses excessive resources and therefore
+needs to operate within the envelope set by network transport circuit
+breaker algorithms.
+
```
Comment (huitema) [inline draft-ietf-ccwg-rfc5033bis.md:362]: How does an endpoint know that a "circuit breaker" is deployed, and what envelope is being used? How wide is the deployment? Are we requiring that CCA developers test with circuit breakers?

Comment (huitema) [reply draft-ietf-ccwg-rfc5033bis.md:362]: RFC8083 is entirely about RTP/RTCP, and the algorithms rely on analyzing the RTCP messages to evaluate packet losses, etc. I think whatever we say here has to be qualified by "if algorithms are designed to control RTP transmissions [cite], they need to remain within the envelope, etc."

**Review by huitema — COMMENTED** [review]
I have a minor comment on BBR (it still is developed by Google), and a more substantial one on "circuit breakers". The circuit breaker specification is entirely focused on RTP/RTCP, and should not be promoted here to apply to the general case. The text should make clear that the circuit breaker section only applies to protocols designed to control RTP/RTCP.

File: draft-ietf-ccwg-rfc5033bis.md (hunk 2 above)
Comment (gorryfair) [reply draft-ietf-ccwg-rfc5033bis.md:362]: I do not understand the comment: if RFC8084 is deployed - then the CCs will be subject to the envelope of this circuit breaker. The text does not explicitly state that they ought to be tested though, maybe it could?  NOTE: that RFC 8084 is NOT about RTP. That topic is dealt with RFC8083, (which is described in the section about multimedia).

File: draft-ietf-ccwg-rfc5033bis.md (hunk 1 above)
Comment (gorryfair) [reply draft-ietf-ccwg-rfc5033bis.md:165]: Hmmm... somebody needs to decide on the form of words. Would this word: " At the time this document was published, BBR is being developed as"....?

File: draft-ietf-ccwg-rfc5033bis.md [hunk 3]
```diff
@@ -346,6 +347,19 @@ full backoff mechanism must be identical to that of TCP
 backoff mechanisms that would give flows with different round-trip
 times comparable capacity during backoff.
 
+### Operation with the Envelope set by Circuit Breakers {{#circuit-breakers}}
+
+A network transport circuit breaker {{!RFC8084}} is an automatic mechanism
+that is used by some equipment in the network to continuously monitor a
+flow or aggregate set of flows.  The mechanism seeks to detect when the
+flow(s) experience persistent excessive congestion, and when detected,
+to terminate (or significantly reduces the rate of) the flow(s).
```
Comment (nealcardwell) [inline draft-ietf-ccwg-rfc5033bis.md:356]: s/reduces/reduce/

File: draft-ietf-ccwg-rfc5033bis.md [hunk 4]
```diff
@@ -186,7 +186,7 @@ Nevertheless, specifying congestion control algorithms has a number of advantage
   which can make it easier for them to suggest improvements and/or identify
   limitations. Further, the specification can help multiple contributors align
   on a consensus change to the algorithm.
-- A specification that is accessible to anyone circumvents the issue that some
+- A specification that is accessible to anyone, can circumvent the issue that some
```
Comment (nealcardwell) [inline draft-ietf-ccwg-rfc5033bis.md:189]: The comma here looks superfluous?

AFAIK, in a statement of the form "An X that is Y can Z" there should not be a comma after the Y...

File: draft-ietf-ccwg-rfc5033bis.md [hunk 5]
```diff
@@ -485,6 +499,18 @@ queues via Differentiated Services Code Points (DSCP) or other mechanisms,
 which can substantially reduce the interplay with other traffic. However, a proposal
 targeting general Internet use can not assume this is always the case.
 
+As noted in {{circuit-breakers}}, all congestion control algorithms
+need to operate within the envelope set by network
+transport circuit breaker algorithms.
+{{!RFC8083}} also defines a
+minimal set of RTP circuit breakers. This identifies conditions under which a sender
+needs to stop transmitting media data to protect the network from excessive congestion.
+It is expected that, in the absence of long-lived excessive congestion
```
Comment (nealcardwell) [inline draft-ietf-ccwg-rfc5033bis.md:508]: Is there a missing comma at the end of this line? I would have expected something like:

  It is expected that, in the absence of long-lived excessive congestion,

File: draft-ietf-ccwg-rfc5033bis.md (hunk 1 above)
Comment (gorryfair) [reply draft-ietf-ccwg-rfc5033bis.md]: At the time of writing, ... has been applied, see latest PR.

File: draft-ietf-ccwg-rfc5033bis.md (hunk 2 above)
Comment (gorryfair) [reply draft-ietf-ccwg-rfc5033bis.md:362]: Saying the level of deployment is a judgment call I'd rather not venture into. I'd be happy though personally to include these in the "special cases" section, if you feel this is more appropriate. 
In reality, the envelope ought not to be crossed by any new algorithm that follows the general rules we laid out, but I do think the original comment is valid: I think the IETF has already provided an envelope and somewhere it ought to appear in this spec.

Comment (huitema) [reply draft-ietf-ccwg-rfc5033bis.md:362]: I think that moving the mention of circuit breakers to a special case is an OK compromise. I did a quick look at citations, and Google Scholar gives 31 results -- which is not bad, but looks to mostly come from IETF drafts and RFC. For comparison, the FQ-Codel spec has 149 citations on Google scholar. It is also referenced in user manuals of commercial routers (e.g., Aruba), while I did not find that for RFC8084. So, some interest, but not at the same level as popular AQM techniques. It does look like a special case.

Comment (gorryfair) [reply draft-ietf-ccwg-rfc5033bis.md]: OK, moved as suggested and framed as a special case.

File: draft-ietf-ccwg-rfc5033bis.md [hunk 6]
```diff
@@ -623,8 +647,8 @@ resource is periodically distributed among the active nodes and where a node mig
 have to buffer data until an assigned transmission opportunity or when the physical path
 changes (e.g., when the length of a wireless path changes, or the physical layer changes
 its mode of operation).
-Variation also arises when a higher priority diffserv traffic classic prompts the
-transmission by a lower class. In these cases, the delay varies as a function of
+Variation also arises when traffic with a higher priority diffserv class pre-empts
+transmission of traffic with a lower class. In these cases, the delay varies as a function of
 external factors and attempting to infer congestion from an increase in the delay
 results in reduced throughput. The jitter from variation over short timescales
 might not be distinguishable similar from other effects.
```
Comment (huitema) [inline draft-ietf-ccwg-rfc5033bis.md:654]: Not part of the PR, but the last sentence in that paragraph is weird. Do we want to say "It might be hard to distinguish the jitter due to variation over short timescales from other effects" ?

File: draft-ietf-ccwg-rfc5033bis.md [hunk 7]
```diff
@@ -37,7 +37,7 @@ author:
 contributor:
  -
     ins: C. Huitema
-    name: Christian Huitema
+    name: Chrtian Huitema
     organization: Private Octopus, Inc.
```
Comment (huitema) [inline draft-ietf-ccwg-rfc5033bis.md:41]: Uh? My name is indeed "Christian Huitema"

File: draft-ietf-ccwg-rfc5033bis.md [hunk 8]
```diff
@@ -82,7 +82,7 @@ informative:
     author:
       ins: Kathleen Nichols
     date: 2011
-    seriesinfo: ACM Queue Volume 9, issue 11
+    seriesinfo: ACM Queue Volume 9, sue 11
```
Comment (huitema) [inline draft-ietf-ccwg-rfc5033bis.md:86]: I think there was an errant edit command replacing "is" by null: Christian -> Chrtian, Lisong Xu -> Long Xu, issue -> sue. Please reverse that!

**Review by huitema — APPROVED** [review]
This looks good, but please don't mangle my name, or that of Lisong Xu, or ACM Queue reference. 

Minor edit suggestion on the wording of the last paragraph in "## Paths with Varying Delay {#delay}"
